unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Tests/PlayMode/SettingsPanelTests_PlayMode.cs

```diff
@@ -0,0 +1,70 @@
+using DCL.SettingsPanelHUD;
+using DCL.SettingsPanelHUD.Sections;
+using NSubstitute;
+using NUnit.Framework;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Tests
+{
+    public class SettingsPanelTests_PlayMode : TestsBase
+    {
+        private const string SECTION_VIEW_PREFAB_PATH = "Sections/DefaultSettingsSectionTemplate";
+        private const string MENU_BUTTON_PREFAB_PATH = "Sections/DefaultSettingsMenuButtonTemplate";
+
+        private SettingsPanelHUDView panelView;
+        private IHUD hudController;
+        private ISettingsPanelHUDController panelController;
+        private SettingsSectionList sectionsToCreate = new SettingsSectionList();
+
+        [UnitySetUp]
+        protected override IEnumerator SetUp()
+        {
+            yield return base.SetUp();
+
+            panelView = SettingsPanelHUDView.Create();
+            hudController = Substitute.For<IHUD>();
+            panelController = Substitute.For<ISettingsPanelHUDController>();
+        }
+
+        protected override IEnumerator TearDown()
+        {
+            Object.Destroy(panelView.gameObject);
+            sectionsToCreate.Clear();
+
+            yield return base.TearDown();
+        }
+
+        [UnityTest]
+        public IEnumerator GenerateSectionsIntoThePanelViewCorrectly()
+        {
+            // Arrange
+            SettingsSectionView sectionViewPrefab = ((GameObject)Resources.Load(SECTION_VIEW_PREFAB_PATH)).GetComponent<SettingsSectionView>();
+            SettingsButtonEntry menuButtonPrefab = ((GameObject)Resources.Load(MENU_BUTTON_PREFAB_PATH)).GetComponent<SettingsButtonEntry>();
+
+            SettingsSectionModel newSectionConfig = new SettingsSectionModel(
+                Sprite.Create(new Texture2D(10, 10), new Rect(), new Vector2()),
```

Comment:
Please make sure the texture/sprite is being freed when the test finishes.

Reply:
Fixed in this new PR (in order to avoid mix things between the part I and II): https://github.com/decentraland/explorer/pull/1784